gift-certificate-data/src/main/java/com/epam/esm/giftcertificatemodule3/config/PersistenceJPAConfig.java

```diff
@@ -0,0 +1,81 @@
+package com.epam.esm.giftcertificatemodule3.config;
+
+import org.springframework.boot.SpringBootConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.core.env.Environment;
+import org.springframework.dao.annotation.PersistenceExceptionTranslationPostProcessor;
+import org.springframework.jdbc.datasource.DriverManagerDataSource;
+import org.springframework.orm.jpa.JpaTransactionManager;
+import org.springframework.orm.jpa.JpaVendorAdapter;
+import org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean;
+import org.springframework.orm.jpa.vendor.HibernateJpaVendorAdapter;
+import org.springframework.transaction.PlatformTransactionManager;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+
+import javax.sql.DataSource;
+import java.util.Objects;
+import java.util.Properties;
+
+@SpringBootConfiguration
+@EnableTransactionManagement
+public class PersistenceJPAConfig {
+
+    private Environment env;
+
+    public PersistenceJPAConfig(Environment env) {
+        this.env = env;
+    }
+
+    @Bean("factory")
+    public LocalContainerEntityManagerFactoryBean entityManagerFactory() {
```

Comment:
This bean is not required. Spring Boot autoconfiguration should handle this behind the scenes based on `spring.datasource` properties in configuration.